docs/chapters/architecture/SWC/software-container.rst

```diff
@@ -0,0 +1,358 @@
+----------------------
+
+The Software Container is a framework that manage and contain applications created
+and vetted by third party developers in an automotive setting. A launcher 
+(a UI for example) sends a signal to the Software Container to start a new 
+Container. It can then configure the Container using a documented JSON format and 
+can launch process inside the Container. This can then be used to contain 
+applications from each other and make sure they are not interfering with each 
+other and to limit/manage resources available to the applications.
+
+Each container is configured using gateways, where each gateway has a scope of 
+influence (network, dbus, files, pulseaudio, for example). The gateways get a 
+configuration snippet from the Launcher which they will enforce.
+
+Requirements overview
+^^^^^^^^^^^^^^^^^^^^^
+
+Basic usage
+"""""""""""
+
+- Software Container Agent component is started at the PELUX boot-up
+- An application launcher is used to launch applications in containerized form 
+  using the d-bus interface provided by the Software Container Agent. In PELUX 
+  QtApplicationManager is used to launch Neptune3 applications in container
+
+Basic requirements
+""""""""""""""""""
+
+Software Container Agent should provide mechanisms,
+
+- To create container
+- To bind-mount filesystem paths
+- To enable communication between the contained system and the host system in 
+  order to access host system environment, services and resources, for e.g.
+
++-----------------------+
+| File System           |
++-----------------------+
+| Devices               |
++-----------------------+
+| Environment variables |
++-----------------------+
+| D-Bus                 |
++-----------------------+
+| Internet Connectivity |
++-----------------------+
+| cgroups               |
++-----------------------+
+
+- To destroy container
+
+Quality goals
+^^^^^^^^^^^^^
+
+The main quality goal for the Software Container is to create sand-box for 
+applications/processes in a way that the applications inside the container can 
+only access the host system's environment, services and resources through the 
+configured capabilities/gateways.
+
+Stakeholders
+^^^^^^^^^^^^
+
+- Software Developer
+- Platform Integrator
+
+System context
+--------------
+
+Business context
+^^^^^^^^^^^^^^^^
+
+Within automotive software, it may be desirable in some cases to utilize a single 
+ECU for several functions. For e.g. it would be beneficial to use the same ECU 
+for Telematics and Infotainment functions, rather than having separate under-utilized 
+ECUs for each. However, applications within these functions can have different 
+resource requirements as well as different levels of importance/priorities, therefore 
+it is important to have mechanisms to separate them in a way that they don't 
+interfere with each other. This can be achieved through the use of Software Container 
+which achieve this goal through,
+
+**Application Isolation**
```

Comment:
Instead of simply using bold text, we should use the header. I believe after the `^^^^^^^^^^` is `"""""""""""""`. In this way they will be visible on the toctree.